fossunited/api/hackathon.py

```diff
@@ -342,3 +342,62 @@ def join_team_via_code(team_code: str, user: str):
 
     team.append("members", {"member": participant.name})
     team.save()
+
+
+@frappe.whitelist()
+def get_session_participant(hackathon: str) -> dict:
+    """
+    Get participant details of the current session user
+
+    Args:
+        hackathon (str): Hackathon ID
+
+    Returns:
+        dict: Participant document as a dictionary
+    """
+    participant = frappe.db.get_value(
+        "FOSS Hackathon Participant",
+        {"hackathon": hackathon, "user": frappe.session.user},
+        [
+            "name",
+            "user_profile",
+            "full_name",
+            "email",
+            "is_student",
+            "git_profile",
+            "organization",
+            "hackathon",
+            "wants_to_attend_locally",
+            "localhost",
+            "localhost_request_status",
+        ],
+        as_dict=1,
+    )
+    return participant
+
+
+@frappe.whitelist()
+def delete_project(hackathon: str, team: str):
+    """
+    Delete team project
+
+    Args:
+        hackathon (str): Hackathon ID
+        team (str): Team ID
+    """
+    team_doc = frappe.get_doc("FOSS Hackathon Team", team)
+    if frappe.session.user not in [
+        member.email for member in team_doc.members
+    ]:
```

Comment:
i'm happy with whatever you folks decide (@RiyaMathew-11 and @harshtandiya ) - whether all members of a team should be able to delete the project or if only the "manager" of the team should be able to delete it

Reply:
If only the owner of the team were to delete the project, then an edge case would be - if a team owner leaves the team, and the team still exists with other members, who would be the new team owner ? - we need an ownership transfer condition there I think.

Reply:
PS: I just tested a case where, there are 2 members in a team I created. I left the team, and then rejoined the team with the same code I had. 

On rejoining - I am still the team owner. Also getting a timestamp mismatch error while leaving a team. 

cc: @harshtandiya 

https://github.com/user-attachments/assets/7b79a3fe-9eb7-459a-b632-d04f0c79d68b